samcli/local/apigw/local_apigw_service.py

```diff
@@ -193,8 +193,9 @@ def _parse_lambda_output(lambda_output, binary_types, flask_request):
         if not isinstance(json_output, dict):
             raise TypeError("Lambda returned %{s} instead of dict", type(json_output))
 
+        headers = LocalApigwService._merge_response_headers(json_output.get("headers"),
```

Comment:
You should set defaults when getting the keys. This will simplify the `_merge_response_headers` method.